torchrl/objectives/common.py

```diff
@@ -73,6 +73,39 @@ def __init__(self):
         self._has_update_associated = False
         self.value_type = self.default_value_estimator
         # self.register_forward_pre_hook(_parameters_to_tensordict)
+        self.tensordict_keys = {}
+
+    def _set_default_tensordict_keys(self, tensordict_keys):
+        """Specify which tensordict keys should be used and can be configured by this loss."""
+        self.tensordict_keys = tensordict_keys
+        self.set_keys(**self.tensordict_keys)
+
+    def _set_deprecated_ctor_keys(self, **kwargs):
+        """Helper function setting a tensordict key and creating a warning for using a deprecated argument."""
+        for key, value in kwargs.items():
+            if value is not None:
+                warnings.warn(
+                    f"Setting '{key}' via ctor is deprecated, use .set_keys({key}='some_key') instead.",
+                    category=DeprecationWarning,
+                )
+                self.tensordict_keys[key] = value
+        self.set_keys(**self.tensordict_keys)
+
+    def set_keys(self, **kwargs):
+        """Specify tensordict key for given argument.
+
+        Examples:
+            >>> from torchrl.objectives import DQNLoss
+            >>> # initialize the DQN loss
+            >>> actor = torch.nn.Linear(3, 4)
+            >>> dqn_loss = DQNLoss(actor, action_space="one-hot")
+            >>> dqn_loss.set_keys(priority_key="td_error", action_value_key="action_value")
+        """
+        for key, value in kwargs.items():
+            if key not in self.tensordict_keys.keys():
+                raise ValueError(f"{key} not a valid tensordict key")
+            set_value = value if value is not None else self.tensordict_keys[key]
+            setattr(self, key, set_value)
```

Comment:
I'd rather have these keys contained in a separate container, like a dictionary or else.
I'm afraid that as the number of keys increase, we'll end up with a class with many attributes and no easy access to the list of them all.

Reply:
No, the container is already there. However, using the keys would be involve always one extra step.
```
tensordict.get(self.action_key) -> tensordict.get(self.tensordict_keys["action_key"])
```
From the data organisation it would be the better solution, but the code will be a little more noisy.

Reply:
I agree it's a bit clunky.
@matteobettini do you have an opinion on the matter?

Reply:
I prefer `tensordict.get(self.tensordict_keys["action_key"])`.

Also, would `loss_keys` make sense instead of `tensordict_keys`

Reply:
What about an Enum instead of a dict (to make sure only a finite set of keys is present)?
```
tensordict.get(self.loss_keys.action_key)
```

Reply:
> What about an Enum instead of a dict (to make sure only a finite set of keys is present)?
This would mean constructing the enum in the base class from data provided by the child class. I guess this is possible, since almost everything is possible in python. However, the syntax would not look like
```
tensordict.get(self.loss_keys.action_key)
```
since the key must be converted to str (or pair of strings). StrEnum would be a solution, but are introduced in python 3.11.